peripherals/adc/adc_f4.yaml

```diff
@@ -0,0 +1,148 @@
+"ADC?":
+  STAT:
+    ROVF:
+      _read:
+        NoOverrun: [0, "No overrun occurred in routine data register"]
+        Overrun: [1, "Overrun overrun occurred in routine data register"]
+      _write:
+        Clear: [0, "Clear the routine data register overrun flag"]
+    STRC:
+      _read:
+        NotStarted: [0, "No regular channel conversion started"]
+        Started: [1, "Regular channel conversion has started"]
+      _write:
+        Clear: [0, "Clear the regular channel start flag"]
```

Comment:
It looks like many of these are copied from `adc_common.yaml`. Please share common peripherals via includes where possible, rather than copying and pasting. See how e.g. `adc_f3.yaml` just includes `adc_common.yaml` and some other files for the parts that are different.

Reply:
Yes, it's true that many of these are common with adc_common.yaml but there are some subtle differences in the f4 series that might take a long time to compare / integrate with adc_common. So this is more of an intermediate step to have things working.